src/openrct2/actions/WallPlaceAction.hpp

```diff
@@ -0,0 +1,651 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../OpenRCT2.h"
+#include "../management/Finance.h"
+#include "../ride/RideGroupManager.h"
+#include "../ride/Track.h"
+#include "../ride/TrackData.h"
+#include "../world/Banner.h"
+#include "../world/LargeScenery.h"
+#include "../world/MapAnimation.h"
+#include "../world/Scenery.h"
+#include "../world/SmallScenery.h"
+#include "../world/Surface.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, GameActionResult)
+{
+private:
+    int32_t _wallType;
```

Comment:
Could default initialize some members to invalid values in case of serialisation failure, I'm planning to get the fuzzer going again on this weekend to see what the game actions are doing when someone would do something stupid.

Reply:
done